llvm/lib/Target/WebAssembly/WebAssemblyRegStackify.cpp

```diff
@@ -260,7 +260,10 @@ static void query(const MachineInstr &MI, bool &Read, bool &Write,
 // Test whether Def is safe and profitable to rematerialize.
 static bool shouldRematerialize(const MachineInstr &Def,
                                 const WebAssemblyInstrInfo *TII) {
-  return Def.isAsCheapAsAMove() && TII->isTriviallyReMaterializable(Def);
+  return Def.isAsCheapAsAMove() && TII->isTriviallyReMaterializable(Def) &&
+         llvm::all_of(Def.all_uses(), [](const MachineOperand &MO) {
+           return MO.getReg().isVirtual();
+         });
```

Comment:
Should put this in a utility function somewhere

Reply:
This particular one is folded away in https://github.com/llvm/llvm-project/pull/160377, but in general, yes.

Reply:
Actually, looking at this again, aren't I missing an isReg() check here?  How is this not failing?